src/app/pages/ValidatorDetailsPage/BalanceDistributionCard.tsx

```diff
@@ -1,13 +1,60 @@
 import { FC } from 'react'
 import { useTranslation } from 'react-i18next'
-import { SnapshotTextCard } from '../../components/Snapshots/SnapshotCard'
+import { TFunction } from 'i18next'
+import { Validator } from '../../../oasis-nexus/api'
+import { getPreciseNumberFormat } from '../../../locales/getPreciseNumberFormat'
+import { SnapshotCard } from '../../components/Snapshots/SnapshotCard'
+import { PieChart } from '../../components/charts/PieChart'
 
-export const BalanceDistributionCard: FC = () => {
+type BalanceDistributionCardProps = {
+  validator?: Validator
+}
+
+function chartData(t: TFunction, validator: Validator | undefined) {
+  if (
+    validator?.escrow?.active_balance === undefined ||
+    validator?.escrow?.self_delegation_balance === undefined
+  ) {
+    return []
+  }
+
+  const activeShares = Number(validator.escrow.active_shares)
+  const selfDelegationShares = Number(validator.escrow.self_delegation_shares)
+  const otherShares = activeShares - selfDelegationShares
+
+  return [
+    {
+      label: t('validator.active'),
+      value: activeShares,
+    },
+    {
+      label: t('validator.self'),
+      value: selfDelegationShares,
+    },
+    {
+      label: t('validator.others'),
+      value: otherShares,
+    },
+  ]
+}
+
+export const BalanceDistributionCard: FC<BalanceDistributionCardProps> = ({ validator }) => {
   const { t } = useTranslation()
 
   return (
-    <SnapshotTextCard title={t('validator.balanceDistribution')}>
-      {/* TODO: provide voting power stats when API is ready */}-
-    </SnapshotTextCard>
+    <SnapshotCard title={t('validator.balanceDistribution')}>
+      <PieChart
+        compact
+        data={chartData(t, validator)}
+        dataKey="value"
+        formatters={{
+          data: (value: number) =>
+            t('common.valueLong', {
+              ...getPreciseNumberFormat(value.toString()),
```

Comment:
This is not precise `TextBigInt` -> `Number(validator.escrow.*)` -> `value.toString()`

Same issue in ConsensusAccountDetailsPage/BalanceDistribution.tsx, reproducible in https://explorer.dev.oasis.io/mainnet/consensus/address/oasis1qpg2xuz46g53737343r20yxeddhlvc2ldqsjh70p